CMakeLists.txt

```diff
@@ -80,9 +80,11 @@ IF(NOT KOKKOS_HAS_TRILINOS)
       SET(CMAKE_CXX_COMPILER ${SPACK_CXX} CACHE STRING "the C++ compiler" FORCE)
       SET(ENV{CXX} ${SPACK_CXX})
     ENDIF()
-  ENDif()
+  ENDIF()
+  # Always call the project command to define Kokkos_ variables
+  # and to make sure that C++ is an enabled langague
+  PROJECT(Kokkos CXX)
   IF(NOT DEFINED PROJECT_NAME)
```

Comment:
With this pull request, the check here is redundant and we are indeed modifying the build type for the parent project here.
We could instead do
```
GET_DIRECTORY_PROPERTY(HAS_PARENT PARENT_DIRECTORY)
IF(NOT HAS_PARENT)
  ...
ENDIF()
```

Reply:
Ha, yes, good catch.